README.md

```diff
@@ -420,6 +420,82 @@ The local key helper is responsible for:
+
+Here is an example of the Local Key Helper XPC service plist:
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+    <key>Label</key>
+    <string>com.contoso.LocalKeyHelper</string>
+    <key>Program</key>
+    <string>/pathToXcpService/LocalKepHelper.xpc/Contents/MacOS/LocalKepHelper</string>
+    <key>RunAtLoad</key>
+    <true/>
+    <key>KeepAlive</key>
+    <true/>
+    <key>MachServices</key>
+    <dict>
+        <key>com.contoso.LocalKeyHelper</key>
+        <true/>
+    </dict>
+</dict>
+</plist>
+```
+
+When the browser (e.g. Chrome) communicates with this service, it needs to establish a service connection first.
+Here is an example of starting a new connection with the Local Key Helper:
+
+```objective-c
+// Start a new connection by providing the Local Key Helper defined in the plist
+NSXPCConnection* localKeyhelper = [[NSXPCConnection alloc] initWithMachServiceName:@"com.contoso.LocalKeyHelper"];
+
+// Make the connection conform to the defined Local Key Helper protocol
+[localKeyhelper setRemoteObjectInterface:[NSXPCInterface interfaceWithProtocol:@protocol(LocalKeyHelper)]];
+```
+
+The browser (e.g. Chrome) can then call the methods defined in the LocalKeyHelper protocol, and the implementations will be provided by the XPC service.
+
+**Note:**
+
+- It is important to define the LocalKeyHelper protocol exactly the same way in both the browser and the XPC service provider. This allows for easy extension of the protocol in the future (e.g. using a dictionary as an input parameter instead of a strict object type).
+- The input and output parameters of the XPC service should follow the `NSSecureCoding` protocol.
+- Un-sandboxed applications can communicate directly to the XPC service.
+
+#### Define Available Local Key Helpers
+To inform the browser about the Local Key Helper to use, the manifest file is created within the browser's root folder during the LocalKeyHelper's installation (e.g. `/Library/Google/Chrome/LocalKeyHelpers`). This folder contains the following files:
+````
+com.provider1.LocalKeyHelper.json (if helper was installed)
+com.provider2.LocalKeyHelper.json (if helper was installed)
+com.provider3.LocalKeyHelper.json (if helper was installed)
+````
+For example:
+````
+com.contoso.LocalKeyHelper.json
+````
+And for each manifest file:
+
+```json
+  {
+    "provider": "contoso",
+      "xpc": // Local key helper type
+      {
+        "service": "com.contoso.LocalKeyHelper", // service name/API activation id
+        "version": "0.0.1", // Helper service version
+        "urls":["login.contoso1.com", "login.contoso2.net"] // This are the URLs the local key helper is used for
```

Comment:
Urls is too generic.
Should we use "idp"?

Reply:
can this list be easily modified by MDM?

Reply:
we also need to support *
urls: [*]
urls: [*.micosoftonline.com]

Reply:
login.microsoft.com
   "com.contoso.LocalKeyHelper"

Reply:
Updated based on our discussion